colorpicker/src/main/java/com/fourmob/colorpicker/ColorPickerPalette.java

```diff
@@ -0,0 +1,162 @@
+
+    private Image createBlankSpace() {
+        Image imageView = new Image(getContext());
+        DirectionalLayout.LayoutConfig layoutParams = new DirectionalLayout
+                .LayoutConfig(this.mSwatchLength, this.mSwatchLength);
+        layoutParams.setMargins(this.mMarginSize, this.mMarginSize, this.mMarginSize, this.mMarginSize);
+        imageView.setLayoutConfig(layoutParams);
+        return imageView;
+    }
+
+    private ColorPickerSwatch createColorSwatch(RgbColor color, RgbColor selectedColor) {
+        ColorPickerSwatch colorPickerSwatch = new ColorPickerSwatch(getContext(),
+                color, color.asArgbInt() == selectedColor.asArgbInt(), this.mOnColorSelectedListener);
+        DirectionalLayout.LayoutConfig layoutParams = new DirectionalLayout
+                .LayoutConfig(this.mSwatchLength, this.mSwatchLength);
+        layoutParams.setMargins(this.mMarginSize, this.mMarginSize, this.mMarginSize, this.mMarginSize);
+        colorPickerSwatch.setLayoutConfig(layoutParams);
+        return colorPickerSwatch;
+    }
+
+    private DirectionalLayout createTableRow() {
+        DirectionalLayout localTableRow = new DirectionalLayout(getContext());
+        localTableRow.setOrientation(Component.HORIZONTAL);
+        localTableRow.setLayoutConfig(new ComponentContainer.LayoutConfig(-2, -2));
+        return localTableRow;
+    }
+
+    private void setSwatchDescription(int rowNumber, int index, int rowElements, boolean selected,
+                                      Component swatch) {
+        int accessibilityIndex;
+        if (rowNumber % 2 == 0) {
+            // We're in a regular-ordered row
+            accessibilityIndex = index;
+        } else {
+            // We're in a backwards-ordered row.
+            int rowMax = ((rowNumber + 1) * mNumColumns);
+            accessibilityIndex = rowMax - rowElements;
+        }
+
+        String description;
+        if (selected) {
+            description = String.format(mDescriptionSelected, accessibilityIndex);
+        } else {
+            description = String.format(mDescription, accessibilityIndex);
+        }
+        swatch.setAccessibilityDescription(description);
+    }
+
+    /**
+     * draw the palette.
+     *
+     * @param colors list of colors to be drawn on palette.
+     * @param selectedColor selected color.
+     */
+    public void drawPalette(RgbColor[] colors, RgbColor selectedColor) {
+        if (colors == null) {
+            return;
+        }
+        this.removeAllComponents();
+        int tableElements = 0;
+        int rowElements = 0;
+        int rowNumber = 0;
+        // Fills the table with swatches based on the array of colors.
+        DirectionalLayout row = createTableRow();
+        for (RgbColor color : colors) {
+            tableElements++;
+            Component colorSwatch = createColorSwatch(color, selectedColor);
+            setSwatchDescription(rowNumber, tableElements, rowElements, color == selectedColor,
```

Comment:
color == selectedColor will always return false, as the reference is different eventhough the values are same. 
Compare the values by doing this "color.asArgbInt() == selectedColor.asArgbInt()"